samples/web_shell/scripts/git.js

```diff
@@ -0,0 +1,319 @@
+    // and a 'message' event listener attached.  This wrapping method is used
+    // instead of attaching the event listeners directly to the <EMBED> element
+    // to ensure that the listeners are active before the NaCl module 'load'
+    // event fires.
+    this.listenerDiv.appendChild(moduleEl);
+  }.bind(this));
+};
+
+GitSalt.prototype.statusText = 'NO-STATUSES';
+
+GitSalt.prototype.updateStatus = function(opt_message) {
+  if (opt_message) {
+    statusText = opt_message;
+  }
+}
+
+/**
+ * Add the default "load" and "message" event listeners to the element with
+ * id "listener".
+ *
+ * The "load" event is sent when the module is successfully loaded. The
+ * "message" event is sent when the naclModule posts a message using
+ * PPB_Messaging.PostMessage() (in C) or pp::Instance().PostMessage() (in
+ * C++).
+ */
+GitSalt.prototype.attachDefaultListeners = function() {
+  this.listenerDiv.addEventListener('load', this.moduleDidLoad.bind(this), true);
+  this.listenerDiv.addEventListener('message', this.handleMessage.bind(this), true);
+  this.listenerDiv.addEventListener('error', this.handleError.bind(this), true);
+  this.listenerDiv.addEventListener('crash', this.handleCrash.bind(this), true);
+};
+
+/**
+ * Called when the NaCl module fails to load.
+ *
+ * This event listener is registered in createNaClModule above.
+ */
+GitSalt.prototype.handleError = function(event) {
+  this.updateStatus('ERROR [' + this.naclModule.lastError + ']');
+};
+
+/**
+ * Called when the Browser can not communicate with the Module
+ *
+ * This event listener is registered in attachDefaultListeners above.
+ */
+GitSalt.prototype.handleCrash = function(event) {
+  if (this.naclModule.exitStatus == -1) {
+    this.updateStatus('CRASHED');
+  } else {
+    this.updateStatus('EXITED [' + this.naclModule.exitStatus + ']');
+  }
+  if (typeof window.handleCrash !== 'undefined') {
+    window.handleCrash(this.naclModule.lastError);
+  }
+};
+
+/**
+ * Called when the NaCl module is loaded.
+ *
+ * This event listener is registered in attachDefaultListeners above.
+ */
+GitSalt.prototype.moduleDidLoad = function() {
+  this.updateStatus('RUNNING');
+  if (typeof window.moduleDidLoad !== 'undefined') {
+    window.moduleDidLoad();
+  }
+  if (this.loadPromise != null) {
+    this.loadPromise();
+    this.loadPromise = null;
+  }
+};
+
+/**
+ * Hide the NaCl module's embed element.
+ *
+ * We don't want to hide by default; if we do, it is harder to determine that
+ * a plugin failed to load. Instead, call this function inside the example's
+ * "moduleDidLoad" function.
+ *
+ */
+GitSalt.prototype.hideModule = function() {
+  // Setting GitSalt.naclModule.style.display = "None" doesn't work; the
+  // module will no longer be able to receive postMessages.
+  this.naclModule.style.height = '0';
+};
+
+/**
+ * Remove the NaCl module from the page.
+ */
+GitSalt.prototype.removeModule = function() {
```

Comment:
I don't see any call to this function, does this mean we leak a git object at each invocation?